### PR TITLE
IGAPP-1136: Fix overlapping POIs

### PR DESCRIPTION
**Pull request by LeandraH**
This pull request belongs to an issue on our [bugtracker](https://issues.integreat-app.de/).
You can find it there by looking for an issue with the key which is mentioned in the title of this pull request.
It starts with the keyword **IGAPP**.

POIs that are very close to each other or at the same address currently only display the first one (I believe in alphabetical order), so other POIs disappear. Here, overlapping ones have the names displayed in different locations. It's still not perfect, e.g. you need to zoom in really far for them to not be clustered anymore (but with a maximum cluster zoom, it displays only one POI again when the cluster disappears).

image

There is a [design ticket](https://issues.tuerantuer.org/browse/IGAPP-1146) for coming up with a better solution but for now, this at least doesn't hide POIs anymore.

### CODE REVIEW

**Review by steffenkleinle — APPROVED** [review]
LGTM, but I think we should wait for a review from @f1sh1918.

File: release-notes/unreleased/IGAPP-1136-fix-overlapping-pois-hiding-each-other.yml
Comment (f1sh1918) [inline release-notes/unreleased/IGAPP-1136-fix-overlapping-pois-hiding-each-other.yml:7]: pls add german translation for play store

Comment (steffenkleinle) [reply release-notes/unreleased/IGAPP-1136-fix-overlapping-pois-hiding-each-other.yml:7]: Only necessary if `show_in_stores` is set to true. But I think it should be set to true so I agree with @f1sh1918 .

**Review by f1sh1918 — CHANGES_REQUESTED** [review]
Nice work, i just found two small bugs

1. When you zoom close enough to see the pois with the same address and the select one, it zooms out and shows them as a cluster. (tested on iOS)
Either we could adjust the `detailZoom` (to 16) or somehow check these Pois and give them a special flag and use a different zoom for them.
2. The marker labels overlaps the clusteringCountLayer. check my comments to fix this

File: release-notes/unreleased/IGAPP-1136-fix-overlapping-pois-hiding-each-other.yml
Comment (f1sh1918) [reply release-notes/unreleased/IGAPP-1136-fix-overlapping-pois-hiding-each-other.yml:7]: oh sorry didn't see that.

File: api-client/src/testing/PoiModelBuilder.ts
Comment (f1sh1918) [inline api-client/src/testing/PoiModelBuilder.ts:30]: pls reverted if change not needed

File: api-client/src/utils/__tests__/geoJson.spec.ts
Comment (f1sh1918) [inline api-client/src/utils/__tests__/geoJson.spec.ts:99]: rm clog

File: api-client/src/utils/geoJson.ts
Comment (f1sh1918) [inline api-client/src/utils/geoJson.ts:23]: its easier and cleaner to use turfs distance function and then set maybe
```js
 // 5 meters
  const maxDistanceForOverlap = 0.05
```
```js
const overlappingCoordinates = coordinateList.filter(coord => distance(coord, coordinates) < maxDistanceForOverlap)
```

**Review by f1sh1918 — COMMENTED** [review]
Added some comments to simplify the calculation

File: api-client/src/maps/index.ts
Comment (f1sh1918) [inline api-client/src/maps/index.ts:32]: maybe i overseen it but is there any reason to make it optional? I think its `false` by default

File: api-client/src/testing/PoiModelBuilder.ts
Comment (LeandraH) [reply api-client/src/testing/PoiModelBuilder.ts:30]: I think it makes it easier to understand to have them in this order because that's the same order as in the coordinates.

File: api-client/src/maps/index.ts
Comment (LeandraH) [reply api-client/src/maps/index.ts]: Good point

**Review by f1sh1918 — APPROVED** [review]
Tested on chrome & ios. Nice work :)

**Pull request merged**
